scripts/helpers.js

```diff
@@ -0,0 +1,75 @@
+// Not intended to be run as a script
+
+var shell = require('shelljs');
+
+var assignments = [];
+var exerciseDirs = shell.ls('-d', 'exercises/*');
```

Comment:
Not a fan of a forced side-effect when loading this script. Can we move this to a function? Memoize it if necessary.

Reply:
> Not a fan of a forced side-effect when loading this script.

Neither am I. But if we make it a function, we have numerous calls like `assignments()[i]` which are not pleasing to the eye. Also since this is a one-time running script, I think this tradeoff is bearable.

> Memoize it if necessary.

Memoization is a side-effect :)

Reply:
It means I _must_ have that directory ready prior to executing any dependants (which may be a sore source of bugs). If the issue is the numerous calls I rather have one of two:
- in the _executing_ scripts use: `const assignments = readAssignments()` or something like that, once.
- export default someObject where you've done `Object.defineProperty('assignments'` and it's a getter, so you can have the syntax you want 🗡 

> Memoization is a side-effect :)

 But not forced until you call the function -- so not an import side-effect and keeps the tree shakable.

That said, I'm fine if you leave it. I understand your reasoning. I also don't like assignments()[i], which is why I would load it into a local var first.